### PR TITLE
Use case class ParsedDeltaSharingTablePath as return type for parsePath

### PR DESCRIPTION
**Pull request by linzhou-db**
Use case class ParsedDeltaSharingTablePath instead of tuple as return type for parsePath, this is part of https://github.com/delta-io/delta-sharing/issues/341

### CODE REVIEW

**Review by charlenelyu-db — APPROVED** [review]
LGTM

File: client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala [hunk 1]
```diff
@@ -841,7 +847,7 @@ object DeltaSharingRestClient extends Logging {
    * Parse the user provided path `profile_file#share.schema.share` to
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala:847]: Update comment?

File: client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala [hunk 2]
```diff
@@ -74,6 +74,12 @@ trait DeltaSharingClient {
   def getProfileProvider: DeltaSharingProfileProvider = null
 }
 
+case class ParsedDeltaSharingTablePath(
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala:77]: Consider documenting this class.

File: client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala [hunk 3]
```diff
@@ -65,6 +65,7 @@ private[sharing] class DeltaSharingFileSystem extends FileSystem {
 
   override def getUri(): URI = URI.create(s"$SCHEME:///")
 
+  // delta-sharing:///
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala:68]: This comment is not clear. Consider expanding on it.

File: client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala (hunk 2 above)
Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala:77]: @chakankardb  added in another PR.

File: client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala (hunk 1 above)
Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala:847]: updated.

File: client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala (hunk 3 above)
Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala:68]: updated.

**Pull request merged**